woocommerce/woocommerce-gutenberg-products-block
# Product Price: support for background color, line height, font family, letter spacing, and padding.

This Pull Request enhances the styling capabilities of the Product Price block by introducing additional support for background color, line height, font family, letter spacing, and padding. This improvement provides users with more flexibility when customizing the appearance of the Product Price block.


Part of #8692 & #7954

### Testing
#### Test using blocks sidebar
1. Create a new page or post
2. Add the "Products" block to your post.
3. Select the Product Price block and open the block settings in the right sidebar.
4. Customize the block using the new styling options available:
     - Change the background color under the "Color" settings.
     - Adjust the line height, font family, and letter spacing under the "Typography" settings.
     - Modify the padding under the "Dimensions" settings.
5. Verify that the styling changes are immediately visible and applied correctly within the editor itself.
6. Save the post, then view it on the front-end to ensure the styling changes are visible and applied correctly.

#### Test using blocks sidebar
1. Open the Global Styles sidebar by clicking the "Styles" button at the top right corner of the editor.

https://user-images.githubusercontent.com/16707866/228781997-24a9a29f-4be7-46d4-b1b3-41520692d86d.mov

2. In the Global Styles sidebar, locate the "Product Price" block.
3. Customize the block using the new styling options available under the "Product Price" block in the Global Styles sidebar:
      - Change the background color under the "Colors" settings.
      - Adjust the line height, font family, and letter spacing under the "Typography" settings.
      - Modify the padding under the "Layout" settings.
4. Verify that these applied correctly within editor & on frontend:
      1. Create a new post
      2. Add Products block
      3. Add Product Price block as an inner block if not included already  
      4. Check the Product Price block in both the editor and on the front-end to ensure that the changes made through the Global styles UI are accurately applied:
        - In the editor, inspect Product Price block & ensure it matches the value set in the Global styles UI.
        - Save the post & publish it. On frontend, inspect Product Price block & ensure it matches the value set in the Global styles UI.

> Also, verify these in `all products` block, which also uses Product Price block.


### WooCommerce Visibility




### Changelog

> Enhance Product Price block styling capabilities with additional support for background color, line height, font family, letter spacing, and padding.
